File: app/src/main/java/io/lerk/lrkFM/entities/HistoryEntry.java
package io.lerk.lrkFM.entities;

import android.support.annotation.Nullable;

/**
 * History entry.
 *
 * @author Lukas Fülling (lukas@k40s.net)
 */
public class HistoryEntry {
    private final FMArchive archive;
    private final String path;
    private final Boolean inArchive;

    public HistoryEntry(String path, @Nullable FMArchive archive) {
        this.path = path;
        this.inArchive = archive != null;
        this.archive = archive;
    }

    public String getPath() {
        return path;
    }

    public Boolean isInArchive() {
        return inArchive;
    }

    @Nullable
    public FMArchive getArchive() {
        return archive;
    }
}

File: app/src/main/java/io/lerk/lrkFM/entities/HistoryEntry.kt
package io.lerk.lrkFM.entities

/**
 * History entry.
 *
 * @author Lukas Fülling (lukas@k40s.net)
 */
class HistoryEntry(val path: String?, val archive: FMArchive?) {
    val isInArchive: Boolean

    init {
        isInArchive = archive != null
    }
}
